pennylane/transforms/condition.py

```diff
@@ -99,6 +98,84 @@ def qnode():
             m_1 = qml.measure(2)
             qml.cond(m_0, qml.RZ)(sec_par, wires=1)
             return qml.expval(qml.PauliZ(1))
+
+    .. UsageDetails::
+
+        **Conditional quantum functions**
+
+        The ``cond`` transform allows conditioning quantum functions too:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=2)
+
+            def qfunc(par, wires):
+                qml.Hadamard(wires[0])
+                qml.RY(par, wires[0])
+
+            @qml.qnode(dev)
+            def qnode():
+                qml.Hadamard(0)
+                m_0 = qml.measure(0)
+                qml.cond(m_0, qfunc)(first_par, wires=[1])
+                return qml.expval(qml.PauliZ(1))
+
+        .. code-block :: pycon
+
+            >>> par = np.array(0.3, requires_grad=True)
+            >>> qnode()
+            tensor(0.45008329, requires_grad=True)

```

Comment:
Maybe worth having `qnode()` take the parameter as an argument, e.g., `qnode(par)`? To avoid users potentially copying and pasting the code, and having odd closure/scope issues :)